### DISCUSSION TITLE
[Frontend/Backend] 회원가입 / 로그인 요청 및 응답 구조

### DISCUSSION THREAD

**Discussion by codero225** [80000Coding/80000Coding-Web-Client#63] [General]
![Untitled Workspace](https://github.com/80000Coding/80000Coding-Web-Client/assets/137995283/dad49f24-87c3-4636-a0a1-8843c3838a6e)

대략적인 sequence diagram을 그려봤습니다.
우선적으로 같이 확인하고 싶은 부분은 아래와 같습니다.

1. OAuth 로그인 후 frontend에서 backend로 authorization code를 넘길 때의 주소
2. backend에서 github user의 정보를 응답받고 존재여부 확인 후 frontend로 넘길 때의 응답 (신규회원 / 기존회원)
3. Authorization code나 Access Token을 github으로 넘길 때 에러가 날 경우의 처리

우선 생각해 본 의견입니다.
1. 어떤 keyword를 쓸 지 안떠오르네요... 의견 부탁드립니다...
2. code로 신규회원인지, 기존회원인지를 나타내고 신규회원이라면 github ID를 담은 JWT를, 기존회원이라면 팔코DB에서 해당 사용자의 정보(id, role 등)를 담은 JWT를 반환
3. authorization code가 잘못되는 경우는 시간이 지났거나 코드가 잘못된 경우인데, 한 번이라도 사용됐다면 잘못된 경우로 봅니다. 처음부터 다시 로그인하게 만들면 좋을 것 같습니다.

잘못된 부분이나 의견을 받고 같이 정하면 좋겠습니다!

**Comment by rkskekzzz** [comment]
1번은 Callback URL을 말씀하시는 걸까용?
2번 동일한 형태로 응답을 보내주시면 좋을 것 같습니다. 코드를 다르게 해서 구별하는 방식으로 하면 어떨까용?
```javascript
{
  code : "SIGNIN_ALLOWED" or "SIGNUP_REQUIRED",
  message : "로그인에 성공했습니다" or "회원가입이 필요합니다"
}
```
3번은 백엔드분들께 맡기겠습니다... ㅎㅎ

**Reply by codero225** [reply]
1번은 callback URL 이후의 이야기 입니다!
깃허브가 callback URL + authorization code를 parameter로 준다음에 back으로 어떻게 요청할 것인가~ 에 대한 내용입니다 ㅎㅎ

**Comment by psychology50** [comment]
3번은 그냥 요청 에러나는 거라, 프론트단에서 예외 처리하면 끝나는 문제입니당
